internal/validations/clusterlogforwarder/validate_clusterlogforwarderspec_test.go

```diff
@@ -120,6 +120,41 @@ var _ = Describe("Validate clusterlogforwarderspec", func() {
 			checkReceiver(&loggingv1.ReceiverSpec{}, `ReceiverSpecs are only supported for the vector log collector`, map[string]bool{})
 		})
 
+		It("should fail validation for invalid Syslog receiver specs", func() {
+
+			checkReceiver := func(receiverSpec *loggingv1.ReceiverSpec, expectedErrMsg string, extras map[string]bool) {
+				const inputName = `syslog-receiver`
+				forwarderSpec := &loggingv1.ClusterLogForwarderSpec{
+					Inputs: []loggingv1.InputSpec{
+						{
+							Name:     inputName,
+							Receiver: receiverSpec,
+						},
+					},
+				}
+				verifyInputs(forwarderSpec, clfStatus, extras)
+				Expect(clfStatus.Inputs[inputName]).To(HaveCondition("Ready", false, loggingv1.ReasonInvalid, expectedErrMsg))
+			}
+
+			checkPorts := func(port int32, expectedErrMsg string) {
+				checkReceiver(
+					&loggingv1.ReceiverSpec{
+						Syslog: &loggingv1.SyslogReceiver{
+							Port: port,
+						},
+					},
+					expectedErrMsg,
+					map[string]bool{constants.VectorName: true},
+				)
+			}
+
+			for _, port := range []int32{-1, 53, 80_000} {
+				checkPorts(port, `invalid port specified for Syslog receiver`)
+			}
+			checkReceiver(&loggingv1.ReceiverSpec{}, `ReceiverSpec must define an Syslog receiver`, map[string]bool{constants.VectorName: true})
+			checkReceiver(&loggingv1.ReceiverSpec{}, `ReceiverSpecs are only supported for the vector log collector`, map[string]bool{})
+		})
+
```

Comment:
need a negative test here for both HTTP and Syslog receivers defined together.

Reply:
i think HTTP and Syslog defined together is not  something we would like to stop

Reply:
Right, we should add this test